permiso/src/main/java/com/greysonparrelli/permiso/PermisoDialogFragment.java

```diff
@@ -48,16 +55,102 @@ public static PermisoDialogFragment newInstance(@Nullable String title, @NonNull
         return dialogFragment;
     }
 
+    public static class Builder {
+        private int titleId = 0;
+        private String title = null;
+        private int buttonTextId = 0;
+        private String message = null;
+        private int messageId = 0;
+        // by default, interpret the dialog msg body as string text
+        private boolean interpretHtml = false;
+        private String buttonText = null;
+
+
+        private Builder() {}
+
+        public Builder(int titleId, int msgBody, int buttonTextId) {
```

Comment:
Is there a reason that the caller must supply all of these resIds when instantiating the builder? I think the main advantage of the builder pattern is that you can specify ids or strings for each of the fields (and you've already added the necessary methods down below). Ideally the constructor would take no args.